logstash-core-event/lib/logstash/string_interpolation.rb

```diff
@@ -125,7 +125,9 @@ def evaluate(event)
       when Hash
         LogStash::Json.dump(value)
       else
-        value
+        # Make sure we dont work on the refence of the value
+        # The Java Event implementation was always returning a string.
+        "#{value}"
```

Comment:
it needs to be a new string instance such that if you upcase! the returned value it does not mutate the original.

In jruby to_s on String returns a reference to the original.